lib/utils/discover.ts

```diff
@@ -46,3 +34,60 @@ export async function discoverLocalBalenaOsDevices(
 			port,
 		}));
 }
+
+async function searchBalenaDevicesOverNetworks(
+	timeout: number,
+): Promise<Service[]> {
+	const networks = os.networkInterfaces();
+	const validNics: os.NetworkInterfaceInfo[] = [];
```

Comment:
It's a bit subjective, but I'd either inline this function into `discoverLocalBalenaOsDevices`'s body, or make this function take `networks` as an argument. Right now, this function doesn't have any reason to exist besides the function name that clearly states what it does.

I'd personally err towards the former (to discourage our future selves to reuse something that is not meant to be reused), with a brief comment that it's doing a search over all networks (which is information currently conveyed by this function's name`)